mantidproject/mantid
# Kafka Event Stream Decoder Json Instrument Parsing

**Description of work.**

This work is related to #26213 #26491 and #26645. This is the final stage of coping with streamed instrument geometry in Kafka.

**To test:**

- Thorough code review
- ensure unit tests pass
- run kafka live tests and ensure instrument geometry is as expected in Mantid.


*There is no associated issue. However related to #26213* 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
